### PR TITLE
Cli: improve vote-account vote-authority display

### LINKED ISSUE #11273: `solana vote-account` authorized-voter display is not clear
#### Problem
We currently display a vote account info on the CLI as follows:

```
$ solana vote-account vote.json
Account Balance: 0.02685864 SOL
Validator Identity: 7hguFUbg1pbkSppuwfZ1i9n9Wrqs22J4nLRepGnjvniW
Authorized Voters: {2: "7hguFUbg1pbkSppuwfZ1i9n9Wrqs22J4nLRepGnjvniW"}
Authorized Withdrawer: 7hguFUbg1pbkSppuwfZ1i9n9Wrqs22J4nLRepGnjvniW
Credits: 0
Commission: 100%
Root Slot: ~
Recent Timestamp: BlockTimestamp { slot: 0, timestamp: 0 }
```

The line `Authorized Voters: {2: "7hguFUbg1pbkSppuwfZ1i9n9Wrqs22J4nLRepGnjvniW"}` is not clear what the two values represents, which is: epoch: authorized_voter. So if you authorize a new voter, it should be reported there, keyed to the epoch it changed.

#### Proposed Solution
 - Change `Authorized Voters` line to just display the pubkey of the auth voter, same format as auth withdrawer.
 - Add a separate line for the epoch change data, like: `Authorized Voter last changed during Epoch: [epoch]`
 - Should it be Authorized Voter, not Authorized Voters?

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
In `solana vote-account` output, the Vote Authority line is not clear. It returns a map of one or more numbers to address strings, eg:
```
Vote Authority: {2: "A1T3QdPZGRT99twji2M4vrkFRDXjp8aEbFc6GCJuviSp"}
```

The number is in fact an epoch, the epoch the respective authority is active. This is usually the current epoch, but in the case of a vote-authority update, it is the epoch in which the new vote authority will go into effect.

Cleaning up small, old Labs issues: https://github.com/solana-labs/solana/issues/11273

#### Summary of Changes
Simplify normal display:
```
Vote Authority: A1T3QdPZGRT99twji2M4vrkFRDXjp8aEbFc6GCJuviSp
```

If a new vote authority has been authorized, display its information, eg:
```
Vote Authority: A1T3QdPZGRT99twji2M4vrkFRDXjp8aEbFc6GCJuviSp
  New Vote Authority as of Epoch 4: EvgdtuXcKoBAMW6eL6KMiqDre1do5o2v4tinFv91H2Fo
```

Fixes https://github.com/solana-labs/solana/issues/11273

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
Also, a question: I think the json output of vote authorities should properly be an array (echoes [this issue](https://github.com/solana-labs/solana/issues/12786)), ie:
```
"authorizedVoters": [
    {
        "epoch": "3"
        "voter": "A1T3QdPZGRT99twji2M4vrkFRDXjp8aEbFc6GCJuviSp",
    },
    {
        "epoch": "4"
        "voter": "EvgdtuXcKoBAMW6eL6KMiqDre1do5o2v4tinFv91H2Fo",
    }
]
```
instead of current:
```
"authorizedVoters": {
    "3": "A1T3QdPZGRT99twji2M4vrkFRDXjp8aEbFc6GCJuviSp",
    "4": "EvgdtuXcKoBAMW6eL6KMiqDre1do5o2v4tinFv91H2Fo"
  }
```

Too breaking to change? (Or maybe okay after v2 version bump?)

File: cli-output/src/cli_output.rs [hunk 1]
```diff
@@ -1648,7 +1648,21 @@ impl VerboseDisplay for CliAuthorizedVoters {}
 
 impl fmt::Display for CliAuthorizedVoters {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{:?}", self.authorized_voters)
+        if let Some((_epoch, current_authorized_voter)) = self.authorized_voters.first_key_value() {
```
Comment (t-nelson) [inline cli-output/src/cli_output.rs:1651]: "surely we can't resolve the curre... why the hell do we do all of this superfluous work for every vote?"

File: cli-output/src/cli_output.rs [hunk 2]
```diff
@@ -1648,7 +1648,21 @@ impl VerboseDisplay for CliAuthorizedVoters {}
...
+            write!(f, "{current_authorized_voter}")?;
+        } else {
+            write!(f, "None")?;
+        }
+        if self.authorized_voters.len() > 1 {
+            let (epoch, upcoming_authorized_voter) =
+                self.authorized_voters.last_key_value().unwrap();
```
Comment (t-nelson) [inline cli-output/src/cli_output.rs]: nit: prefer `expect("why?")`

**Review by t-nelson — DISMISSED** [review]
while part of me wishes it didn't, lgtm :sweat_smile:

File: cli-output/src/cli_output.rs (hunk 2 above)
Comment (CriesofCarrots) [reply cli-output/src/cli_output.rs]: Fixed!

File: cli-output/src/cli_output.rs (hunk 1 above)
Comment (CriesofCarrots) [reply cli-output/src/cli_output.rs:1651]: It is a good question!

**Review by t-nelson — APPROVED** [review]

**Pull request merged**